packages/frontend-2/components/server-management/DeleteProjectDialog.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <LayoutDialog
+    v-model:open="isOpen"
+    max-width="sm"
+    :title="title"
+    :buttons="dialogButtons"
+  >
+    <div class="flex flex-col gap-6">
+      <p>
+        Are you sure you want to
+        <strong>permanently delete</strong>
+        the selected project?
+      </p>
+      <div v-if="project">
+        <strong>{{ project.name }}</strong>
+        <p>
+          {{ project.models.totalCount }} models,
+          {{ project.versions.totalCount }} versions,
+        </p>
+      </div>
+      <p>
+        This action
+        <strong>cannot</strong>
+        be undone.
+      </p>
+    </div>
+  </LayoutDialog>
+</template>
+
+<script setup lang="ts">
+import { LayoutDialog } from '@speckle/ui-components'
+import { ProjectItem } from '~~/lib/server-management/helpers/types'
+import { useMutation } from '@vue/apollo-composable'
+import {
+  convertThrowIntoFetchResult,
+  getCacheId,
+  getFirstErrorMessage,
+  updateCacheByFilter
+} from '~~/lib/common/helpers/graphql'
+import { useGlobalToast, ToastNotificationType } from '~~/lib/common/composables/toast'
+import { adminDeleteProject } from '~~/lib/server-management/graphql/mutations'
+import { getProjects } from '~~/lib/server-management/graphql/queries'
+import { Exact, InputMaybe } from '~~/lib/common/generated/gql/graphql'
+
+const props = defineProps<{
+  title: string
+  open: boolean
+  project: ProjectItem | null
+  resultVariables:
```

Comment:
_(I originally posted this in DeleteInvitationDialog, but I've already fixed that one so I'm posting this here)_

This is a bit smelly, you shouldn't need to feed in a specific set of variables just for the mutation cache update. Additionally this isn't going to fully achieve what you're after anyway, because your cache update function will only update the cache for these variables, but any other instances of the `inviteList` query (e.g. with a different search query) will remain stale.

Here I console logged out the Apollo cache and you can see there are multiple instances of the `inviteList` query:
<img width="852" alt="image" src="https://github.com/specklesystems/speckle-server/assets/938316/b9fdc980-c6ed-4bd5-bbee-deb106daf7ab">

A better option is to use `cache.modify` or the util I created - `modifyObjectFields` which essentially wraps `cache.modify`, because it allows you to update multiple variations of the same field. So you can essentially update all instances of the `inviteList` query with a single call, without needing to know the variables beforehand, which is perfect for this scenario.

The complication here is that `AdminInviteList` is not uniquely identifiable - it doesn't have an ID field. And thus you can't just directly update all inviteLists like so: 

```js
cache.modify({
  id: 'AdminInviteList', // there's no ID 
  fields: {
    totalCount: (old) => old + -1
  }
})
```

What you can do, however, is identify `ROOT_QUERY` which is the root of the Query graph. And from there you can modify its `admin` field, which will in turn contain all variations of the `inviteList` field.

This is a bit complicated, so I've already done this for you in the invites page, but you'll need to do the same in other pages as well.

See the `modifyObjectFields` call in `DeleteInvitationDialog.vue` for an example. Cache updates can be complicated, so if something doesn't work right you can utilize the browser debugger to see what's going on.

Reply:
Make sure you fix this in all of your dialogs